query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -106,8 +119,17 @@ final case class TsCardReduceExec(queryContext: QueryContext,
   override protected def compose(childResponses: Observable[(QueryResult, Int)],
                                  firstSchema: Task[ResultSchema],
                                  querySession: QuerySession): Observable[RangeVector] = {
-    val taskOfResults = childResponses.flatMap(res => Observable.fromIterable(res._1.result))
-      .foldLeftL(new mutable.HashMap[ZeroCopyUTF8String, CardCounts])(mapFold)
+
+    // capture the result schema of responses
+    var rs = ResultSchema.empty
+    val flatMapData = childResponses.flatMap(res => {
+      if (rs == ResultSchema.empty) {
+        rs = res._1.resultSchema
```

Comment:
do all child responses have the same schema?

Reply:
Yep, for the same version, each of them will have same schema